lib/Minion/Backend/Pg.pm

```diff
@@ -222,22 +222,28 @@ sub retry_job {
 }
 
 sub stats {
-  my $self = shift;
+  my ($self, $queue) = (shift, shift // '');
 
   my $stats = $self->pg->db->query(
-    "SELECT COUNT(*) FILTER (WHERE state = 'inactive' AND (expires IS NULL OR expires > NOW())) AS inactive_jobs,
+    q{SELECT COUNT(*) FILTER (WHERE state = 'inactive' AND (expires IS NULL OR expires > NOW())) AS inactive_jobs,
        COUNT(*) FILTER (WHERE state = 'active') AS active_jobs, COUNT(*) FILTER (WHERE state = 'failed') AS failed_jobs,
        COUNT(*) FILTER (WHERE state = 'finished') AS finished_jobs,
        COUNT(*) FILTER (WHERE state = 'inactive' AND delayed > NOW()) AS delayed_jobs,
        (SELECT COUNT(*) FROM minion_locks WHERE expires > NOW()) AS active_locks,
        COUNT(DISTINCT worker) FILTER (WHERE state = 'active') AS active_workers,
        (SELECT CASE WHEN is_called THEN last_value ELSE 0 END FROM minion_jobs_id_seq) AS enqueued_jobs,
-       (SELECT COUNT(*) FROM minion_workers) AS inactive_workers,
+       (SELECT COUNT(*) FROM minion_workers WHERE
+          $1 = '' OR EXISTS (SELECT 1 FROM jsonb_array_elements(status->'queues') AS q WHERE q = to_jsonb($1::TEXT))
+       ) AS inactive_workers,
        EXTRACT(EPOCH FROM NOW() - PG_POSTMASTER_START_TIME()) AS uptime
-     FROM minion_jobs"
+     FROM minion_jobs
+     WHERE ($1 = '' OR queue = $1)}, $queue
   )->hash;
   $stats->{inactive_workers} -= $stats->{active_workers};
-
+  if ($queue) {
+    delete $stats->{active_locks};
+    delete $stats->{enqueued_jobs};
+  }
```

Comment:
Doesn't look like this has test coverage.

Reply:
Should it test that `delete` in perl works properly or what kind of coverage is needed here? I guess the change also needs to be addressed in documentation, but that depends on what functionality is wanted. I think returning `active_locks` and `enqueued_jobs` for particular `queue` may be confusing, because those stats are are only for the global level. Am I right? Maybe we should introduce `queue_stats` with required string parameter instead?